src/pkg/packager/packager2.go

```diff
@@ -29,28 +34,68 @@ func getPopulatedVariableConfig(ctx context.Context, pkg v1alpha1.ZarfPackage, s
 	return variableConfig, nil
 }
 
-func generateValuesOverrides(chart v1alpha1.ZarfChart, componentName string, variableConfig *variables.VariableConfig, valuesOverridesMap map[string]map[string]map[string]interface{}) (map[string]any, error) {
-	valuesOverrides := make(map[string]any)
+type overrideOpts struct {
+	variableConfig     *variables.VariableConfig
+	values             value.Values
+	valuesOverridesMap ValuesOverrides
+}
+
+// generateValuesOverrides generates a map of values to override for a given chart and component, with precedence of:
+// Zarf Variable overrides -> Zarf value overrides -> direct API helm-value overrides.
+func generateValuesOverrides(ctx context.Context, chart v1alpha1.ZarfChart, componentName string, opts overrideOpts) (map[string]any, error) {
 	chartOverrides := make(map[string]any)
+	valuesOverrides := make(map[string]any)
 
 	for _, variable := range chart.Variables {
-		if setVar, ok := variableConfig.GetSetVariable(variable.Name); ok && setVar != nil {
+		if setVar, ok := opts.variableConfig.GetSetVariable(variable.Name); ok && setVar != nil {
 			// Use the variable's path as a key to ensure unique entries for variables with the same name but different paths.
 			if err := helpers.MergePathAndValueIntoMap(chartOverrides, variable.Path, setVar.Value); err != nil {
 				return nil, fmt.Errorf("unable to merge path and value into map: %w", err)
 			}
 		}
 	}
 
+	// Map ChartValues' Source to Target
+	for _, chartValue := range chart.Values {
+		if chartValue.SourcePath == "" || chartValue.TargetPath == "" {
+			return nil, fmt.Errorf("sourcePath \"%s\" and targetPath \"%s\" must not be empty", chartValue.SourcePath, chartValue.TargetPath)
+		}
+		if chartValue.SourcePath[0] != '.' {
+			return nil, fmt.Errorf("sourcePath \"%s\" must start with a dot", chartValue.SourcePath)
+		}
+		if chartValue.TargetPath[0] != '.' {
+			return nil, fmt.Errorf("targetPath \"%s\" must start with a dot", chartValue.TargetPath)
+		}
+
+		// Extract value from source path in values
+		sourceValue, err := opts.values.Extract(value.Path(chartValue.SourcePath))
+		if err != nil {
+			// Log warning but don't fail - source path might not exist
+			logger.From(ctx).Warn("unable to extract value from path",
+				"path", chartValue.SourcePath,
+				"error", err,
+				"component", componentName,
+				"chart", chart.Name,
+			)
+			continue
+		}
+
+		// Strip off the leading dot from the target path.
+		targetPath := chartValue.TargetPath[1:]
```

Comment:
A single dot should be valid for both `sourcePath` and `targetPath`. Would be good to get a test for this behavior. Ref: https://github.com/zarf-dev/proposals/blob/main/0021-zarf-values/README.md#proposal

Reply:
IMO it's worth adding in this PR. I see it as a bug in the behavior.

Reply:
That's fair - as I explored a solution to this more, this ended up being one of those problems that once you solve it a bunch of other things click into place.

I made sure both extract and set operations on Values support root paths. This ended up enabling some reuse and simplification in several places Values are used - both helm overrides and SetValues parsing with `values.Set` and the overall merge functionality. I also went over some places where unit tests were missing, so the behavior should be covered on a few system layers.